### DISCUSSION TITLE
Merging pre-sorted files

### DISCUSSION THREAD

**Discussion by evansd** [duckdb/duckdb#10252] [General]
I have data spread across number of Parquet files, each of which are already sorted. I want to import these into a single DuckDB table, retaining the sort order. As the input files are already sorted this can theoretically be done quite efficiently by merging and doesn't require reading all the data first. However, because DuckDB doesn't _know_ that the files are already sorted it can't apply this kind of optimisation.

It would be great if DuckDB supported some way of merging pre-sorted data. Or perhaps there's already some trick to achieving this which I'm missing?

Thanks!

**Comment by hawkfish** [comment]
If we implemented #8444 then we could have a variant of `UNION ALL` that preserved ordering which would accomplish this.

**Reply by szarnyasg** [reply]
@hawkfish did you mean #8444 ?

**Comment by soerenwolfers** [comment]
Is what you're looking for the `POSITIONAL JOIN` https://duckdb.org/docs/sql/query_syntax/from.html#positional-joins
